RFC/RFT: implement taskserver behind SSL proxy

osctobe opened the pull request:
This patchset allows to put taskserver behind an SSL reverse proxy (eg. stunnel or haproxy). Original client's address is passed using PROXY protocol described at [1]. Both versions of the protocol are supported, though binary (v2) is preferred.

[1] http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt

osctobe committed:
TLS: push cert/key config down to Server class

osctobe committed:
TLS: split TLS-specific connection handling

osctobe committed:
cmake: update required version to fix warning about 'test' target

osctobe committed:
TCP: fix accept() for IPv6 clients

osctobe committed:
PROXY: implement PROXYv1 protocol

osctobe committed:
diag: prevent exceptions on missing file checks

osctobe committed:
Server: fix catching string exceptions

aleprovencio commented:
So this should explain why I was never able to put `taskserver` behind `traefik`, right?

Any reasons why this has not been merged yet??

jrabbit commented:
> 
> 
> So this should explain why I was never able to put `taskserver` behind `traefik`, right?
> 
> Any reasons why this has not been merged yet??

taskd expects to be able to terminate the mTLS session itself. You might be able to passively proxy it but you can't frontload it I don't think, without this PR.

osctobe committed:
TLS: split TCP handling

Note: this removes 'chunked' receive and associated buffer copies
trading it for allocating whole receive buffer up front.

osctobe committed:
PROXY: implement PROXYv2 protocol
